tracing/span.go

```diff
@@ -292,6 +297,31 @@ func (s *Span) preStop(err error) {
 	}
 }
 
+// getHub returns the *sentry.Hub attached to this span/trace.
+//
+// It's guaranteed to be non-nil.
+func (s Span) getHub() *sentry.Hub {
+	if s.hub != nil {
+		return s.hub
+	}
+	// This shouldn't happen, but just in case to avoid panics.
+	return getNopHub()
+}
+
+// InjectSentryHub injects the sentry hub attached to this span to the context
+// object as the current hub.
+//
+// It's called automatically by StartSpanFromHeaders and thriftbp/httpbp
+// middlewares,
+// so you don't need to call it for spans created automatically from requests.
+// But you should call it if you created a top level span manually.
+//
+// It's also not needed to be called for the child spans,
+// as the hub attached would be the same.
+func (s Span) InjectSentryHub(ctx context.Context) context.Context {
```

Comment:
🔕 Do we actually want to inject the hub on the context or just export the code to get it from the span?

Reply:
This is needed by the new `log.ErrorWithSentry` API: https://github.com/reddit/baseplate.go/pull/154/files#diff-fbf850c1cb9fc6234051c4e9517a5274R240

If we don't inject it into the context, then we need to add span into that API instead (or try to extract span out of the context), that way this is less useful, as injecting hub into context is a pattern already provided by sentry-go API.